confluent_server/confluent/auth.py

```diff
@@ -115,6 +115,47 @@ class PromptsNeeded(Exception):
     def __init__(self, prompts):
         self.prompts = prompts
 
+ #add function to change _allowedbyrole and _deniedbyrole vars.
+ def add_roles(dictionary):
+    #function to parse the roles and the files. If there are modifications to be done to the roles, items will be added to dictionaries.
+    #If there are no moodifications done to one of the roles, it continues to the next
+    #Opening YAML file and reading the custom roles
+    with open("/etc/confluent/authorization.yaml","r") as stream:
+        loaded_file = yaml.safe_load(stream)
+        for outside_key,outside_value in loaded_file.items():
+            for inside_key,inside_value in outside_value.items():
+                try:
+                    #Trying to append the new list of permissions to existing lists (i.e. Operator : {"retrieve" : ['*' , 'new_added_file_permission']}) 
+                    dictionary[outside_key][inside_key] = (list(set(dictionary[outside_key][inside_key]+inside_value)))
+                except KeyError:
+                    #If there is no previous action, we create a new one (i.e. Operator : { "new_action" : ['new_added_file_permission'] })
+                    try:
+                        dictionary[outside_key][inside_key] = inside_value
+                    except KeyError:
+                        #If there is a new role to be added, we add it along with the rest of the info (i.e. NewRole : {"new_action" : ['new_added_file_permission]})
+                        dictionary[outside_key] = outside_value
+                
+    
+def check_for_yaml():
+    #impot yaml and op.path to check if the file exists and to safe_load the yaml file.
+    try:
+        import yaml
```

Comment:
Go ahead and import yaml from the onset.  There are other parts of confluent that already expect yaml